### PR TITLE
Allow escape sequences in WDL strings

### PR DESCRIPTION
**Pull request by scottfrazer**
The parser how has a hook for every time it matches a string-literal token.  It will replace all escape codes (e.g. `\n`, `\r`, `\u266d`).

Going the other way, `WdlString.toWdlString` will now output the escape sequences.

### CODE REVIEW

**Comment by geoffjentry** [comment]
Is this something which has actually come up or just foresight?

**Comment by scottfrazer** [comment]
Yes, it came up in two places:

* Vivek tried to use these, though I still have to make a test that exercises the exact case he tried
* I was writing a test scatter workflow and not having this means that we can't have strings with `"` or `'` characters in it.  I found that out the hard way!

**Comment by geoffjentry** [comment]
hah. ok, thanks

**Comment by mcovarr** [comment]
Can we merge this?

**Comment by scottfrazer** [comment]
Actually I meant to add one more test because I'm not sure if vivek's use case is specifically covered by this code yet.

**Comment by geoffjentry** [comment]
:+1:

File: src/main/resources/grammar.hgr [hunk 1]
```diff
@@ -41,13 +39,13 @@ grammar {
 
     # TODO: these should probably be partials...
```
Comment (mcovarr) [inline src/main/resources/grammar.hgr]: If a partial is a thing that would prevent duplication of the now-hairy regexps, then yes.  :smile:

Comment (scottfrazer) [inline src/main/resources/grammar.hgr]: Yeah, I haven't quite finished "enums as partials" feature in Hermes yet, but when I find some time I'll implement that.  Continuous improvement!

**Comment by scottfrazer** [comment]
https://github.com/scottfrazer/hermes/pull/52 was spawned from discussion Miguel and I had about allowing the parser to specify imports so the grammar file was simpler and could utilize library code.

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**